packages/cli/src/commands/join.ts

```diff
@@ -153,42 +174,87 @@ packageVersion: string
     tags,
     potentialConflicts,
     tagsPrefix,
-    componentsPrefix
+    componentsPrefix,
   }: JoinDocumentContext) {
-    const xTagGroups = 'x-tagGroups';
-    const Tags = 'tags';
-    if (!joinedDef.hasOwnProperty(Tags)) { joinedDef[Tags] = []; }
-    if (!joinedDef.hasOwnProperty(xTagGroups)) { joinedDef[xTagGroups] = []; }
-    if (!potentialConflicts.tags.hasOwnProperty('all')) { potentialConflicts.tags['all'] = {}; }
-    if (!joinedDef[xTagGroups].some((g: any) => g.name === entrypointFilename)) {
-      joinedDef[xTagGroups].push({ name: entrypointFilename, tags: [] });
+    if (!joinedDef.hasOwnProperty(Tags)) {
+      joinedDef[Tags] = [];
+    }
+    if (!potentialConflicts.tags.hasOwnProperty('all')) {
+      potentialConflicts.tags['all'] = {};
+    }
+    if (skipTagsCheck && !potentialConflicts.tags.hasOwnProperty('description')) {
+      potentialConflicts.tags['description'] = {};
     }
-    const indexGroup = joinedDef[xTagGroups].findIndex((item: any) => item.name === entrypointFilename);
-    if (!joinedDef[xTagGroups][indexGroup].hasOwnProperty(Tags)) { joinedDef[xTagGroups][indexGroup][Tags] = []; }
     for (const tag of tags) {
       const entrypointTagName = addPrefix(tag.name, tagsPrefix);
       if (tag.description) {
         tag.description = addComponentsPrefix(tag.description, componentsPrefix!);
       }
-      if (!joinedDef.tags.find((t: any) => t.name === entrypointTagName)) {
+
+      const tagDuplicate = joinedDef.tags.find((t: Oas3Tag) => t.name === entrypointTagName);
+
+      if (tagDuplicate && skipTagsCheck) {
```

Comment:
Do we cover case when `tagsDuplicate && !skipTagsCheck`?

Reply:
Currently there is no sense for this, because we want add `tag` into  `tags` that not yet exist in `joinDef`.